File: app/src/main/java/com/omar/retromp3recorder/app/usecase/ShareUC.java
package com.omar.retromp3recorder.app.usecase;

import com.omar.retromp3recorder.app.repo.FileNameRepo;

import javax.inject.Inject;

import io.reactivex.Completable;

public class ShareUC {
    private final FileNameRepo repo;

    @Inject
    public ShareUC(FileNameRepo repo) {
        this.repo = repo;
    }

    public Completable execute() {
        return Completable.complete();
    }
}

File: app/src/main/java/com/omar/retromp3recorder/app/usecases/ShareUC.kt
package com.omar.retromp3recorder.app.usecases

import com.omar.retromp3recorder.app.modules.share.Sharer
import io.reactivex.Completable
import javax.inject.Inject

class ShareUC @Inject constructor(private val sharingModule: Sharer) {
    fun execute(): Completable {
        return sharingModule.share()
    }
}
